src/include/kdbtypes.h

```diff
@@ -85,7 +89,7 @@ typedef unsigned char             kdb_octet_t;
 typedef signed short              kdb_short_t;
 typedef unsigned short            kdb_unsigned_short_t;
 
-#if SIZEOF_LONG == 4
+#if (LONG_MAX == +2147483647)
```

Comment:
It's possible that I take the route to fix the cmake issue.

IMHO a standards based approach like ANSI C's limits.h content appears much more straight forward to me. That is even independent of build systems.